apps/client/src/app/components/paywall/PaywallDialog.tsx

```diff
@@ -1,62 +1,47 @@
 import { Close } from '@mui/icons-material';
-import { Dialog, DialogTitle, IconButton } from '@mui/material';
+import { useWaitForCheckoutCompleted, ProvisionStatus } from '@stigg/react-sdk';
 import { useState } from 'react';
-import { SuccessfulPlanProvision } from './SuccessfulPlanProvision';
+import {
+  CloseButton,
+  StyledDialog,
+  StyledDialogTitle,
+} from './PaywallDialog.style';
+import { PlanProvisionStatus } from './PlanProvisionStatus';
 import { Paywall } from './StiggPaywall';
-import { useWaitForCheckoutCompleted } from '../../hooks/useWaitForCheckoutCompleted';
 
 type TodosPaywallProps = {
   paywallIsOpen: boolean;
   onClose: () => void;
 };
 
 export function PaywallDialog({ paywallIsOpen, onClose }: TodosPaywallProps) {
-  const [showProvisionSuccess, setShowProvisionSuccess] = useState(false);
-  const { isAwaitingCheckout } = useWaitForCheckoutCompleted(
-    setShowProvisionSuccess
-  );
+  const [provisionStatus, setProvisionStatus] =
```

Comment:
the `useWaitForCheckoutCompleted` exposes the `provisionStatus` as well, so isn't it redundant to handle additional state here?

I guess that it's for having `null` option - so if it's the case, maybe the `useWaitForCheckoutCompleted` can return this as well

Reply:
I thought about that but the issue is that we are not always going through the checkout flow (upgrade/free/trial), so in those cases we want to handle the state by ourself.
I really thought how we can consolidate that but currently I couldn't come up with a better solution for that

Reply:
What you refer to is this code - right?
```
<Paywall
          onSuccessProvision={() =>
            setProvisionStatus(ProvisionStatus.SUCCEEDED)
          }
        />
```

Reply:
Looked a bit about it, and played with it.
WDYT about:
```
export function PaywallDialog({ paywallIsOpen, onClose }: TodosPaywallProps) {
  const [ hadSuccessProvision, setHadSuccessProvision ] = useState(false);
  const { checkoutProvisionStatus } = useWaitForCheckoutCompleted();

  const provisionStatus = hadSuccessProvision ? ProvisionStatus.SUCCESS : checkoutProvisionStatus;
  const open = paywallIsOpen || !!provisionStatus;

  return (
    <StyledDialog open={open} onClose={onClose} fullWidth>
      <StyledDialogTitle>
        <CloseButton onClick={onClose}>
          <Close />
        </CloseButton>
      </StyledDialogTitle>

      {provisionStatus ? (
        <PlanProvisionStatus provisionStatus={provisionStatus} />
      ) : (
        <Paywall onSuccessProvision={() => setHadSuccessProvision(true)} />
      )}
    </StyledDialog>
  );
}
```

Reply:
I see 👍 

Maybe after the task that @roystigg is working on to unify `initiateCheckout` and `createSubscription` into a single mutation, it will simplify things here.